commit 4d8a8dceafd9632e2d2405880d70f9b8b4409720
Author: Alexander Sosedkin <monk@unboiled.info>
Date:   Wed Sep 29 13:53:59 2021 +0200

    feat: hide select branch names in gitstatus

diff --git a/xonsh/environ.py b/xonsh/environ.py
index a2cc6e25..1e19f404 100644
--- a/xonsh/environ.py
+++ b/xonsh/environ.py
@@ -1404,6 +1404,10 @@ class PromptSetting(Xettings):
         ),
         pattern="XONSH_GITSTATUS_*",
     )
+    XONSH_GITSTATUS_BRANCHES_HIDDEN = Var.with_default(
+        (),
+        "Branch names to hide in {gitstatus} prompt.",
+    )
     XONSH_GITSTATUS_FIELDS_HIDDEN = Var.with_default(
         (),
         "Fields to hide in {gitstatus} prompt (all fields below are shown by default.) \n\n"
diff --git a/xonsh/prompt/gitstatus.py b/xonsh/prompt/gitstatus.py
index 6bc886bb..acbb2af0 100644
--- a/xonsh/prompt/gitstatus.py
+++ b/xonsh/prompt/gitstatus.py
@@ -155,6 +155,9 @@ def _get_status_fields():
                 branch = line
             else:
                 branch, rest = line.split("...")
+                hidden = XSH.env.get("XONSH_GITSTATUS_BRANCHES_HIDDEN")
+                if hidden and branch in hidden:
+                    branch = ""
                 if " " in rest:
                     divergence = rest.split(" ", 1)[-1]
                     divergence = divergence.strip("[]")
@@ -236,7 +239,7 @@ def gitstatus_prompt():
 
     ret = ""
     for fld in (_DEFS.BRANCH, _DEFS.AHEAD, _DEFS.BEHIND, _DEFS.OPERATION):
-        if not _is_hidden(fld):
+        if not _is_hidden(fld) and fld:
             val = fields[fld]
             if not val:
                 continue
