diff --git a/src/hydra-queue-runner/build-remote.cc b/src/hydra-queue-runner/build-remote.cc
index 6c647d91..e7495f33 100644
--- a/src/hydra-queue-runner/build-remote.cc
+++ b/src/hydra-queue-runner/build-remote.cc
@@ -528,7 +528,7 @@ void State::buildRemote(ref<Store> destStore,
         }
 
         /* Register the outputs of the newly built drv */
-        if (settings.isExperimentalFeatureEnabled("ca-derivations")) {
+        if (settings.isExperimentalFeatureEnabled(Xp::CaDerivations)) {
           auto outputHashes = staticOutputHashes(*localStore, *step->drv);
           for (auto & [outputId, realisation] : builtOutputs) {
               // Register the resolved drv output
diff --git a/src/hydra-queue-runner/queue-monitor.cc b/src/hydra-queue-runner/queue-monitor.cc
index 8f65699d..11775998 100644
--- a/src/hydra-queue-runner/queue-monitor.cc
+++ b/src/hydra-queue-runner/queue-monitor.cc
@@ -473,7 +473,7 @@ Step::ptr State::createStep(ref<Store> destStore,
           missing.insert({{outputHash, outputName}, maybeOutputPath.second});
         }
       } else {
-        settings.requireExperimentalFeature("ca-derivations");
+        settings.requireExperimentalFeature(Xp::CaDerivations);
         if (!destStore->queryRealisation(DrvOutput{outputHash, outputName})) {
           valid = false;
           missing.insert({{outputHash, outputName}, std::nullopt});
@@ -489,7 +489,7 @@ Step::ptr State::createStep(ref<Store> destStore,
         for (auto & [i, maybePath] : missing) {
             if ((maybePath && localStore->isValidPath(*maybePath)))
                 avail++;
-            else if (settings.isExperimentalFeatureEnabled("ca-derivations") && localStore->queryRealisation(i)) {
+            else if (settings.isExperimentalFeatureEnabled(Xp::CaDerivations) && localStore->queryRealisation(i)) {
                 maybePath = localStore->queryRealisation(i)->outPath;
                 avail++;
             } else if (useSubstitutes && maybePath) {

